src/dst/src/LoadBalancer.cc

```diff
@@ -103,6 +119,64 @@ void LoadBalancer::getNextWorker(ip::address& ip, unsigned short& port)
   }
 }
 
+void LoadBalancer::lookUpWorkers(
+    const char* domain,
+    unsigned short port)  // LoadBalancer & balancer)
+{
+  asio::io_service ios;
+  std::vector<worker> workers_set;
+  udp::resolver::query resolver_query(
+      domain, std::to_string(port), udp::resolver::query::numeric_service);
+  udp::resolver resolver(ios);
+  while (alive) {
+    std::vector<worker> new_workers;
+    boost::system::error_code ec;
+    auto it = resolver.resolve(resolver_query, ec);
+    if (ec)
+      logger_->warn(utl::DST,
+                     203,
+                     "Workers domain resolution failed with error code = {}. "
+                     "Message = {}.",
+                     ec.value(),
+                     ec.message());
+    int new_workers_count = 0;
+    udp::resolver::iterator it_end;
+    for (; it != it_end; ++it) {
+      auto discovered_worker = worker(it->endpoint().address(), port, 0);
+      if (std::find(workers_set.begin(), workers_set.end(), discovered_worker)
+          == workers_set.end()) {
+        workers_set.push_back(discovered_worker);
+        new_workers.push_back(discovered_worker);
+        new_workers_count += 1;
+      }
+    }
+
+    if (new_workers_count == 0) {
+      debugPrint(
+          logger_,
+          utl::DST,
+          "load_balancer",
+          1,
+          "Discovered 0 new workers with the given domain. Total workers = {}.",
+          workers_set.size());
+    } else {
+      debugPrint(logger_,
+                 utl::DST,
+                 "load_balancer",
+                 1,
+                 "Discovered {} new workers with the given domain. Total "
+                 "workers = {}.",
+                 new_workers_count,
+                 workers_set.size());
+    }
+
+    for (auto worker : new_workers)
+      this->addWorker(worker.ip.to_string(), worker.port);
```

Comment:
why this

Reply:
If new workers are discovered on the network, they should be added to the list of workers available for the load balancer. For instance, if the load balancer is launched before some of the workers were up.

Reply:
No I mean why "this->" 
just call addWorker...